### PR TITLE
ghcjs: properly wrap binaries in environment

### PR DESCRIPTION
**Pull request by cstrahan**
This fixes [the problem](https://github.com/NixOS/nixpkgs/pull/7719#issuecomment-99578013) that @dnhgff brought to my attention. Without this patch, `ghcjs --make` can't find the packages installed by Nix.

**Another consideration:**

I believe we ought to have a separate set of env vars for GHCJS, so programs compiled via GHCJS find the correct libs/compiler. This would entail editing [`pkgs/development/haskell-modules/ghc-paths-nix.patch`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/haskell-modules/ghc-paths-nix.patch) so that the the env vars queried are predicated on the compiler. A simple ~~`#ifdef __GHCJS__`~~ `#ifdef ghcjs_HOST_OS` should do.

@peti how does that sound? If you agree, I'll add that to this PR.

### CODE REVIEW

**Comment by peti** [comment]
IMHO, it's fine to make the change to `ghc-paths-nix.patch`; it sounds reasonable to me.

File: pkgs/development/haskell-modules/with-packages-wrapper.nix [hunk 1]
```diff
@@ -62,15 +62,17 @@ buildEnv {
       chmod u+w "$out/bin"
     fi
 
-    for prg in ghc ghci ghc-${ghc.version} ghci-${ghc.version}; do
+    for prg in ${ghcCommand} ${ghcCommand}i ${ghcCommand}-${ghc.version} ${ghcCommand}-${ghc.version}; do
```
Comment (dnhgff) [inline pkgs/development/haskell-modules/with-packages-wrapper.nix]: the code misses an "i" in the second `${ghcCommand}-${ghc.version}`

**Comment by cstrahan** [comment]
So, I ended up just adding a new patch file, that way changes for ghcjs don't cause rebuilds for everyone else.

**Comment by cstrahan** [comment]
I think this is 99% done, but I want to first see if I can send a patch upstream to fix `ghcjs-run`.

**Comment by cstrahan** [comment]
Alright, I think I'm going to do the `ghcjs-run` patch later.

@peti This should be ready to merge now.

/cc @dnhgff

**Comment by dnhgff** [comment]
@cstrahan thanks, yes i've been using this without any problems

**Comment by cstrahan** [comment]
@dnhgff sweet - thanks for the initial report and the follow up :).

**Comment by cstrahan** [comment]
Just added another commit to set the `NIX_GHCJS_*` env vars in the `env` helper, so now this works too:

``` bash
nix-shell -E '((import <nixpkgs> {}).haskell-ng.packages.ghcjs.reflex.env)'
```

**Pull request merged**